src/Avalonia.Base/AvaloniaObject.cs

```diff
@@ -440,7 +455,16 @@ public void SetCurrentValue<T>(StyledProperty<T> property, T value)
             VerifyAccess();
             ValidatePriority(priority);
 
-            return _values.AddBinding(property, source, priority);
+            if (source is IBinding2 b)
+            {
+                if (b.Instance(this, property, null) is not UntypedBindingExpressionBase expression)
+                    throw new NotSupportedException("Binding returned unsupported IBindingExpression.");
+                return GetValueStore().AddBinding(property, expression);
+            }
```

Comment:
Looks like priority is ignored in this code path. Is it expected?

Reply:
Fixed.